service/matching/taskQueueManager.go

```diff
@@ -493,8 +505,8 @@ func (c *taskQueueManagerImpl) renewLeaseWithRetry() (taskQueueState, error) {
 	err := backoff.Retry(op, persistenceOperationRetryPolicy, common.IsPersistenceTransientError)
 	if err != nil {
 		c.metricScope().IncCounter(metrics.LeaseFailurePerTaskQueueCounter)
-		if c.errShouldUnload(err) {
-			c.engine.unloadTaskQueue(c.taskQueueID)
+		if fsb == emitFatalSignal && c.errShouldUnload(err) {
+			c.signalFatalProblem(c.taskQueueID)
```

Comment:
in the long term, the release task queue from matching engine should be moved out, e.g. renewLeaseWithRetry should only renew release, caller should make decision what to do if operation fails

as a temp fix, this PR is ok

Reply:
I agree with this - it should always have been the choice of the caller.